### ISSUE TITLE
Scrollwheel skips every other hotbar item

### ISSUE THREAD

**Issue by tauin** [Admicos/minecraft-wayland#38]
using GNOME Wayland on the 1.19 libdecor branch (issue occurs on MC 1.18 too)

Tested on a superflat world using PrismLauncher, in Xwayland/X11 using the scrollbar to change hotbar items correctly switches between every single hotbar item (1 -> 2 -> 3 -> 4 -> 5 -> 6 etc.) while on the wayland branch it skips every other hotbar item (1 -> 3 -> 6 -> 9 -> 2 -> 4 -> 6 -> 8 -> 1 -> 3 etc.)

**Comment by scaledteam** [comment]
I wrote a very dirty solution, seems to work for me. Also scrolling from touchpad through hotbar is now has a perfect sensitivity.

```
--- a/src/input.c
+++ b/src/input.c
@@ -341,7 +341,7 @@
     assert(yoffset < FLT_MAX);
 
     if (window->callbacks.scroll)
-        window->callbacks.scroll((GLFWwindow*) window, xoffset/2, yoffset/2);
+        window->callbacks.scroll((GLFWwindow*) window, xoffset, yoffset);
 }
 
 // Notifies shared code of a mouse button click event
 ```

**Comment by Neubulae** [comment]
I also have this issue

**Comment by Neubulae** [comment]
> I wrote a very dirty solution, seems to work for me. Also scrolling from touchpad through hotbar is now has a perfect sensitivity.
> 
> ```
> --- a/src/input.c
> +++ b/src/input.c
> @@ -341,7 +341,7 @@
>      assert(yoffset < FLT_MAX);
>  
>      if (window->callbacks.scroll)
> -        window->callbacks.scroll((GLFWwindow*) window, xoffset, yoffset);
> +        window->callbacks.scroll((GLFWwindow*) window, xoffset/2, yoffset/2);
>  }
>  
>  // Notifies shared code of a mouse button click event
> ```

I decided to add this patch to myself and it didn't seem to work for me...? Wonder why

**Comment by scaledteam** [comment]
I applied that patch to this version of glfw: https://github.com/mattfbacon/glfw-wayland-minecraft-git
And also, since i use Debian, not Arch, i patched sources manually, + installed manually into flatpak. Maybe i missed something, but i can't properly test it.

Idea is, that you need to edit "input.c" file at 344 line and divide xoffset and yoffset in 2, so final function will look like:
```
void _glfwInputScroll(_GLFWwindow* window, double xoffset, double yoffset)
{
    assert(window != NULL);
    assert(xoffset > -FLT_MAX);
    assert(xoffset < FLT_MAX);
    assert(yoffset > -FLT_MAX);
    assert(yoffset < FLT_MAX);

    if (window->callbacks.scroll)
        window->callbacks.scroll((GLFWwindow*) window, xoffset/2, yoffset/2);
}
```

**Comment by Neubulae** [comment]
> I applied that patch to this version of glfw: [mattfbacon/glfw-wayland-minecraft-git](https://github.com/mattfbacon/glfw-wayland-minecraft-git) And also, since i use Debian, not Arch, i patched sources manually, + installed manually into flatpak. Maybe i missed something, but i can't properly test it.
> 
> Idea is, that you need to edit "input.c" file at 344 line and divide xoffset and yoffset in 2, so final function will look like:
> 
> ```
> void _glfwInputScroll(_GLFWwindow* window, double xoffset, double yoffset)
> {
>     assert(window != NULL);
>     assert(xoffset > -FLT_MAX);
>     assert(xoffset < FLT_MAX);
>     assert(yoffset > -FLT_MAX);
>     assert(yoffset < FLT_MAX);
> 
>     if (window->callbacks.scroll)
>         window->callbacks.scroll((GLFWwindow*) window, xoffset/2, yoffset/2);
> }
> ```
Yeah this is what I did, but it still skips every other hotbar, dunno which part went wrong
